backend/gateway/stats/runtime.go

```diff
@@ -0,0 +1,145 @@
+)
+
+type RuntimeStatCollector struct {
+	runtimeMetrics runtimeMetrics
+}
+
+type runtimeMetrics struct {
+	cpuGoroutines tally.Gauge
+	cpuCgoCalls   tally.Gauge
+
+	memoryAlloc   tally.Gauge
+	memoryTotal   tally.Gauge
+	memorySys     tally.Gauge
+	memoryLookups tally.Gauge
+	memoryMalloc  tally.Gauge
+	memoryFrees   tally.Gauge
+
+	memoryHeapAlloc    tally.Gauge
+	memoryHeapSys      tally.Gauge
+	memoryHeapIdle     tally.Gauge
+	memoryHeapInuse    tally.Gauge
+	memoryHeapReleased tally.Gauge
+	memoryHeapObjects  tally.Gauge
+
+	memoryStackInuse       tally.Gauge
+	memoryStackSys         tally.Gauge
+	memoryStackMSpanInuse  tally.Gauge
+	memoryStackMSpanSys    tally.Gauge
+	memoryStackMCacheInuse tally.Gauge
+	memoryStackMCacheSys   tally.Gauge
+
+	memoryOtherSys tally.Gauge
+
+	memoryGCSys        tally.Gauge
+	memoryGCNext       tally.Gauge
+	memoryGCLast       tally.Gauge
+	memoryGCPauseTotal tally.Gauge
+	memoryGCCount      tally.Gauge
+}
+
+func NewRuntimeStats(scope tally.Scope) *RuntimeStatCollector {
+	runtimeScope := scope.SubScope("runtime")
+	runtimeMetrics := runtimeMetrics{
+		// cpu
+		cpuGoroutines: runtimeScope.Gauge("cpu.goroutines"),
+		cpuCgoCalls:   runtimeScope.Gauge("cpu.cgo_calls"),
+		// memory
+		memoryAlloc:   runtimeScope.Gauge("memory.alloc"),
+		memoryTotal:   runtimeScope.Gauge("memory.total"),
+		memorySys:     runtimeScope.Gauge("memory.sys"),
+		memoryLookups: runtimeScope.Gauge("memory.lookups"),
+		memoryMalloc:  runtimeScope.Gauge("memory.malloc"),
+		memoryFrees:   runtimeScope.Gauge("memory.frees"),
+		// heap
+		memoryHeapAlloc:    runtimeScope.Gauge("memory.heap.alloc"),
+		memoryHeapSys:      runtimeScope.Gauge("memory.heap.sys"),
+		memoryHeapIdle:     runtimeScope.Gauge("memory.heap.idle"),
+		memoryHeapInuse:    runtimeScope.Gauge("memory.heap.inuse"),
+		memoryHeapReleased: runtimeScope.Gauge("memory.heap.released"),
+		memoryHeapObjects:  runtimeScope.Gauge("memory.heap.objects"),
+		// stack
+		memoryStackInuse:       runtimeScope.Gauge("memory.stack.inuse"),
+		memoryStackSys:         runtimeScope.Gauge("memory.stack.sys"),
+		memoryStackMSpanInuse:  runtimeScope.Gauge("memory.stack.mspan_inuse"),
+		memoryStackMSpanSys:    runtimeScope.Gauge("memory.stack.sys"),
+		memoryStackMCacheInuse: runtimeScope.Gauge("memory.stack.mcache_inuse"),
+		memoryStackMCacheSys:   runtimeScope.Gauge("memory.stack.mcache_sys"),
+		memoryOtherSys:         runtimeScope.Gauge("memory.othersys"),
+		// GC
+		memoryGCSys:        runtimeScope.Gauge("memory.gc.sys"),
+		memoryGCNext:       runtimeScope.Gauge("memory.gc.next"),
+		memoryGCLast:       runtimeScope.Gauge("memory.gc.last"),
+		memoryGCPauseTotal: runtimeScope.Gauge("memory.gc.pause_total"),
+		memoryGCCount:      runtimeScope.Gauge("memory.gc.count"),
+	}
+
+	return &RuntimeStatCollector{
+		runtimeMetrics: runtimeMetrics,
+	}
+}
+
+func (r *RuntimeStatCollector) Collect(ctx context.Context) {
+	ticker := time.NewTicker(time.Second * 10)
```

Comment:
should this interval be configurable?

Reply:
Sure I have no objections to that, ill rework the config a bit.